src/rbr/RBR.java

```diff
@@ -134,22 +134,28 @@ private void adjustRegions(Vertex sw) {
         ArrayList<Region> newRegions = new ArrayList<>();
         ArrayList<Region> regionsToBeRemoved = new ArrayList<>();
         for (Region currentRegion : regionsForVertex.get(sw)) {
-            Set<Vertex> outsiders = currentRegion.outsiders();
-            if(outsiders.isEmpty())
-                continue;
-            Range outsidersBox = TopologyKnowledge.box(outsiders);
-            Set<Vertex> trulyDestinationsInOutsidersRange = currentRegion.destinationsIn(outsidersBox);
-
+            newRegions.addAll(adjustRegions(currentRegion));
             regionsToBeRemoved.add(currentRegion);
-            ArrayList<Region> regionsToAdd = splitRegionExcludingOutsiders(currentRegion, outsidersBox);
-            newRegions.addAll(regionsToAdd);
-            // use others routers to make others regions
-            newRegions.addAll(makeRegions(trulyDestinationsInOutsidersRange, currentRegion.inputPorts(), currentRegion.outputPorts()));
         }
         regionsForVertex.get(sw).removeAll(regionsToBeRemoved);
         regionsForVertex.get(sw).addAll(newRegions);
```

Comment:
The way that you made the extraction 'regionsToBeRemoved' will contain all sw's regions (therefore the statement at 140 will empty the list 'regionsForVertex.get(sw)') and 'newRegions' will contain all regions to be used thereafter. Instead of emptying the list and then filling it completely you could replace the whole list with only one statement. That way you'll no more need 'regionsToBeRemoved'.